File: calc.cpp
#include <iostream>

using namespace std;

float addition(float x, float y){
    return x + y;
}

float subtraction(float x, float y){
    return x - y;
}

float multiplication(float x, float y){
    return x * y;
}

float division(float x, float y){
    if(y == 0){
        cout << "Can not divide by zero" << endl;
    }
    return x / y;
}


int main(){
    float num1, num2, result;
    char what_arithmetic;

    cout << "Enter first number: ";
    cin >> num1;
    if(cin.fail()){
        cout << "Invalid input" << endl;
        cin.clear();
        cin.ignore(1000, '\n');
        return 1;
    }

    cout << "Enter arithmetic: ";
    cin >> what_arithmetic;

    cout << "Enter second number: ";
    cin >> num2;
    if(cin.fail()){
        cout << "Invalid input" << endl;
        cin.clear();
        cin.ignore(1000, '\n');
        return 1;
    }

    switch (what_arithmetic)
    {
    case '+':
        result = addition(num1, num2);
        cout << "Result is " << result << endl;
        break;
    
    case '-':
        result = subtraction(num1, num2);
        cout << "Result is " << result << endl;
        break;
    
    case '*':
        result = multiplication(num1, num2);
        cout << "Result is " << result << endl;
        break;

    case '/':
        result = division(num1, num2);
        if(num2 != 0){
            cout << "Result is " << result << endl;
            break;
        }else{
            cout << "Can not divide by zero!" << endl;
            break;
        }
    default:
        cout << "Invalid arithmetic" << endl;
        break;
    }
}

File: calc.py
print("Welcome to the calculator!")

def subtraction(x, y):
    return x - y

def addition(x, y):
    return x + y

def multiplication(x, y):
    return x * y

def division(x, y):
    if x or y == 0:
        return "Error: Can not divide by zero"
    return x / y

operations = {
    1: ("Addition", addition),
    2: ("Subtraction", subtraction),
    3: ("Multiplication", multiplication),
    4: ("Division", division)
}

while(True):
    try:

        what_calc = int(input("Press 1 for addition\n"
                        "Press 2 for subtraction\n"
                        "Press 3 for division\n"
                        "Press 4 for multiplication\n"
                        "Press 5 to exit the calculator: "))
        
        if(what_calc == 5):
            print("Come back soon!")
            break
        elif what_calc not in operations:
            print("Invalid input.")
        
        num1 = int(input("Num 1: "))
        num2 = int(input("Num 2: "))
        
        operation_name, operation_func = operations[what_calc]
        result = operation_func(num1, num2)
        print(f"result of {operation_name} is {result}")

    except ValueError:
        print("Values must be of type integer")